L1Trigger/TrackFindingTMTT/src/L1KalmanComb.cc

```diff
@@ -0,0 +1,1828 @@
+///=== This is the base class for the Kalman Combinatorial Filter track fit algorithm.
+
+///=== Written by: S. Summers, K. Uchida, M. Pesaresi, I.Tomalin
+
+#include "L1Trigger/TrackFindingTMTT/interface/L1KalmanComb.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Utility.h"
+
+#include <TMatrixD.h>
+#include "L1Trigger/TrackFindingTMTT/interface/TP.h"
+#include "L1Trigger/TrackFindingTMTT/interface/KalmanState.h"
+#include "L1Trigger/TrackFindingTMTT/interface/StubCluster.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include <algorithm>
+#include <functional>
+#include <fstream>
+#include <iomanip>
+#include <TH2F.h>
+//#define CKF_DEBUG
+// Enable debug printout to pair with that in Histos.cc enabled by recalc_debug.
+//#define RECALC_DEBUG
+
+// Enable merging of nearby stubs.
+//#define MERGE_STUBS
+
+namespace TMTT {
+
+  unsigned LayerId[16] = {1, 2, 3, 4, 5, 6, 11, 12, 13, 14, 15, 21, 22, 23, 24, 25};
+
+  static bool orderStubsByLayer(const Stub *a, const Stub *b) { return (a->layerId() < b->layerId()); }
+
+#ifdef MERGE_STUBS
+  static bool orderStubsByZ(const Stub *a, const Stub *b) { return (a->z() < b->z()); }
+
+  static bool orderStubsByR(const Stub *a, const Stub *b) { return (a->r() < b->r()); }
+#endif
```

Comment:
```suggestion
namespace {

  constexpr unsigned LayerId[16] = {1, 2, 3, 4, 5, 6, 11, 12, 13, 14, 15, 21, 22, 23, 24, 25};

  bool orderStubsByLayer(const Stub *a, const Stub *b) { return (a->layerId() < b->layerId()); }

#ifdef MERGE_STUBS
  bool orderStubsByZ(const Stub *a, const Stub *b) { return (a->z() < b->z()); }

  bool orderStubsByR(const Stub *a, const Stub *b) { return (a->r() < b->r()); }
#endif
}

namespace TMTT {
```
the `LayerId` variable name should also start with lower case letter, and it might be clearer at the use sites to have some prefix/postfix to denote these are global

Reply:
Functions orderStubsByZ() and orderStubsByR() & array LayerId have been eliminated by removing some rarely used code options. Function orderStubsByLayer() has been replaced by a lambda function inside the function where it is used, removing it from the global name space.